### PR TITLE
upgrade to rails 7

### PR DESCRIPTION
**Pull request by peetucket**
Note: trying to switch to codeclimate and circleci in a separate PR before merging this

## Why was this change made?

Upgrade to rails 7

Caveats:
- renamed a single view file to match style of others
- view specs couldn't infer template locations anymore and needed to be told
- i couldn't figure out how to get the single view spec that renders json to work, skipped it  (we have a controller spec already verifying it renders json)

## How was this change tested?

Test suite

### CODE REVIEW

File: spec/models/batch_context_spec.rb [hunk 1]
```diff
@@ -200,13 +200,13 @@
     it 'adds error if missing manifest.csv' do
       allow(File).to receive(:exist?).with('spec/test_data/images_jp2_tif/manifest.csv').and_return(false)
       bc.send(:verify_bundle_directory)
-      expect(bc.errors.to_h).to include(bundle_dir: /missing manifest/)
+      expect(bc.errors.map(&:type)).to include('missing manifest: spec/test_data/images_jp2_tif/manifest.csv')
```
Comment (peetucket) [inline spec/models/batch_context_spec.rb:203]: `to_h` was removed from ActiveRecord::Errors class

**Review by jcoyne — COMMENTED** [review]
Did you run `bin/rails app:update`?

**Comment by peetucket** [comment]
> Did you run `bin/rails app:update`?

I didn't realize this existed.  Going through that now...

File: config/environments/production.rb [hunk 2]
```diff
@@ -16,53 +18,50 @@
...
   # Use a different cache store in production.
   # config.cache_store = :mem_cache_store
 
-  # Use a real queuing backend for Active Job
-  config.active_job.queue_adapter = :resque
+  # Use a real queuing backend for Active Job (and separate queues per environment).
+  # config.active_job.queue_adapter     = :resque
+  # config.active_job.queue_name_prefix = "pre_assembly_production"
 
-  config.action_mailer.default_url_options = { host: Settings.mailer_host, protocol: 'https' } # needed by url_for() in mail templates
```
Comment (jcoyne) [inline config/environments/production.rb:65]: Isn't this important?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: config/environments/production.rb (hunk 2 above)
Comment (peetucket) [reply config/environments/production.rb:65]: yup, the `app:update` script removed them (also from dev and test, which I added back in, but missed this one)

**Comment by peetucket** [comment]
Note: Coveralls seems broken (reporting 0% coverage)

**Review by jcoyne — APPROVED** [review]

**Comment by peetucket** [comment]
I'm going to try and switch to circle and codeclimate in a separate PR.  May hold off on merging this for now (try and rebase after that's done to make sure it's all working here too)

**Pull request merged**